### ISSUE TITLE
Launch sugar in a window

### ISSUE THREAD

**Issue by westurner** [sugarlabs/sugar#929]
> Is there a way to launch sugar in a window (in gnome) at least for development (instead of by selecting sugar as an alternate window manager/launcher in the GDM login screen)? That would solve for: screensaver with lock screen, sugar not working with e.g. LightDM on old notebooks, development and testing.
>
> A person could require the window to launch in fullscreen in kiosk mode if that's necessary.

https://github.com/sugarlabs/sugar/issues/927

**Comment by quozl** [comment]
Needs more user experience detail.  Sugar could be changed to check for command line argument `--no-fullscreen` and;
* not use a full-screen window,
* not start the metacity window manager,
* provide user interface elements for the global keys like F6k,
* not log out the session on exit.

**Comment by westurner** [comment]
I'll take a look at https://github.com/sugarlabs/sugar-runner .

"Kiosk / single-application mode"
https://github.com/sugarlabs/sugar-runner/issues/7

**Comment by devmrfitz** [comment]
Is this issue about running the traditional Sugar session manager view in a window in another manager like Gnome, XFCE, etc?

**Comment by quozl** [comment]
Yes, this issue originally asked about how to run Sugar inside another window manager.  It is an idea that has merit, but has not been implemented in Sugar yet.  I gave items that begin to show what would have to change inside Sugar to complete the idea.

However, there was no progress on defining what was to change or to change it.  Instead something that is not part of Sugar, called Sugar Runner, which used to do something like this but not as well, and is unmaintained, was suggested as a workaround.

**Comment by devmrfitz** [comment]
Should I try to implement it using Xephyr?

**Comment by quozl** [comment]
No.  As in https://github.com/sugarlabs/sugar/issues/929#issuecomment-657303717, implementing support for running Sugar as a window requires a set of steps, and none of them would require Xephyr.  Sugar Runner only used Xephyr and Xvfb in order to _not_ change Sugar.  Sugar Runner was not intended for regular use, but rather as a debugging tool.  Sugar Runner is not a good example of how to implement the changes needed.
